src/rgw/rgw_op.cc

```diff
@@ -590,7 +590,14 @@ void rgw_add_to_iam_environment(rgw::IAM::Environment& e, std::string&& key, std
 }
 
 void rgw_add_to_iam_environment(rgw::IAM::Environment& e, const std::string& key, const std::string& val){
-  e[key] = val;
+  if (!val.empty())
+    e[key] = val;
+}
+
+
+void rgw_add_to_iam_environment(rgw::IAM::Environment& e, const char* key, const char* val){
+  if (key != nullptr && val != nullptr)
```

Comment:
You're checking to see if the pointers are nullptr, but not *also* that the values they contain are empty as above.

Reply:
dropping the empty check, since empty values though rare are legal since policy allows for null conditionals which just check the presence of a key and not a value

Reply:
Did you also drop the empty check from the std::string& version?

Reply:
yes this is dropped now